FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -106,19 +106,29 @@ class Flagsmith constructor(
         const val DEFAULT_ANALYTICS_FLUSH_PERIOD_SECONDS = 10
     }
 
-    fun getFeatureFlags(identity: String? = null, result: (Result<List<Flag>>) -> Unit) {
+    fun getFeatureFlags(identity: String? = null, traits: List<Trait>? = null, result: (Result<List<Flag>>) -> Unit) {
         // Save the last used identity as we'll refresh with this if we get update events
         lastUsedIdentity = identity
 
         if (identity != null) {
-            retrofit.getIdentityFlagsAndTraits(identity).enqueueWithResult { res ->
-                flagUpdateFlow.tryEmit(res.getOrNull()?.flags ?: emptyList())
-                result(res.map { it.flags })
+            if (traits != null) {
+                retrofit.postTraits(IdentityAndTraits(identity, traits)).enqueueWithResult(result = {
+                    result(it.map { response -> response.flags })
+                }).also { lastUsedIdentity = identity }
+            } else {
+                retrofit.getIdentityFlagsAndTraits(identity).enqueueWithResult { res ->
+                    flagUpdateFlow.tryEmit(res.getOrNull()?.flags ?: emptyList())
+                    result(res.map { it.flags })
+                }
```

Comment:
Similar to my comment [here](https://github.com/Flagsmith/flagsmith-ios-client/pull/59#pullrequestreview-2204500269), I think we can remove this conditional logic and always use the `POST` endpoint, but if it's non-trivial to achieve, we can ignore for the purpose of this PR.